Buevest docs2006

BueVest opened the pull request:
First shot at solving #473 and #202 .
Comments or changes welcome.

bertfrees commented on tests/yaml/capitalization.yaml:
```diff
@@ -9,7 +9,7 @@ table: |
   uplow Zz 6
   sign - 23
-  seqdelimiter -
+  seqdelimiter - # Has no effect with capitals
```
What do you mean with this comment? That the could be removed?

BueVest replied:
Yes, I removed it, and it had no effect. Perhaps, it is not because of seqdelimiter having no effect, but rather whatever is wrong with capsphrase.

bertfrees replied:
I don't think there is anything wrong with `capsphrase`. Besides it is not even used in this test.

Let's just remove this line from the test.

I've looked a bit at the code and it seems `seqdelimiter` is not used for applying emphasis, except in one place but I suspect that is a mistake.

bertfrees commented:
Very good additions for the UEB specific opcodes. Another improvement could be to explain a bit more what "standing alone" means practically. You already say that:

> A "standing alone" sequence can be thought of as a word in a very broad sence.

and later you say:

> This again determines whether word contractions, part word contractions or nocontractsign should be applied.

but I'm still not sure I know what exactly the consequences are. Does it mean that it changes the behavior of the `word` opcode? The documentation of the `word` opcode is:

> Replace the characters with the dot pattern if they are a word, that is, are surrounded by whitespace and/or punctuation. 

For the capitalization opcodes, because these are almost completely the same as the emphasis opcodes, I was actually thinking about making the documentation for these very short and refer to their "emph" counterparts, and explain the differences if needed.

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+but may have unpredicted side-effects if used outside the intended
+context. Use with great care, and test thoroughly.
+
+@end table
```
what table are we closing here?

BueVest replied:
The table that was started in line 647.

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+@node Opcodes for Standing Alone Sequences
+@section Opcodes for Standing Alone Sequences
+
+The term "standing alone" comes from the specification of Unified
```
Maybe we should emphasize the term standing alone see the [texinfo manual](https://www.gnu.org/software/texinfo/manual/texinfo/html_node/_0040emph-_0040strong.html)), e.g. like this
``` texinfo
@emph{standing alone}
```
Actually come to think of it, maybe we should even add it to the concept index. Add the following before the paragraph that you want to link in the index:
``` texinfo
@cindex Standing alone
```

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+
+The term "standing alone" comes from the specification of Unified
+English Braille (UEB). In Liblouis, a letter or letters-sequence is
+considered to be "standing alone" if it is preceded and followed by a
```
emphasis on term

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+English Braille (UEB). In Liblouis, a letter or letters-sequence is
+considered to be "standing alone" if it is preceded and followed by a
+space, and/or other characters that you choose as delimiters, e.g.
+@samp{-}. A "standing alone" sequence can be thought of as a word in
```
emphasis on term

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+Spaces do not need to be listed as
+they are automatically delimiters.
+
+For example, in UEB (section 2.6.1 page 15), any hyphen or dash count
```
I was hoping we could have URLs here,  but it appears that the spec is only available as a PDF and BRF

bertfrees replied:
The references are good enough I think.

egli replied:
yes, since we can't link to an URL it is ok as it is

egli commented on doc/liblouis.texi:
```diff
@@ -835,6 +851,99 @@ numericmodechars .,
+
+Example:
+@example
+seqbeforechars ([@{"“'‘
```
These chars have special meaning to texinfo, so they need to be quoted, see [Special Characters: Inserting @ {} , \ # &](https://www.gnu.org/software/texinfo/manual/texinfo/html_node/Special-Characters.html)

egli replied:
ah, sorry I believe you already took care  of the quoting

egli commented:
Oh, I forgot one thing: You often say "Note: something important". In other places of the documentation we use 
``` texinfo
@strong{Note:} 
```
Maybe we should use that here as well.

BueVest commented:
The references as well as some of the other text has been copied straight from the wiki.

BueVest commented:
To tell the truth, I also was and still am a little in doubt, but that is how I read tests/ueb/input-ueb-02-stand_alone.txt (why not yaml tests?)

I will try to test some more with and without seqdelimiter and explain it more fully, perhaps quote an example from the UEB tests.

 

Concerning capitalization, I think it would be ok to make a reference to the emphasis doc, but it is a larger and more complicated subject than caps. So, I think the doc of caps should still be “standing alone”. There is some discussion in the wiki about the difference between caps and emphasis, but I am not sure it ever comes to a real conclusion.

bertfrees commented:
> This again determines whether word contractions, part word contractions or nocontractsign should be applied.

I just discovered in the code that `seqdelimiter` only has an effect on `word` and `contraction` (and `nocontractsign`).

bertfrees commented:
> So, I think the doc of caps should still be “standing alone”.

I just think that if we want to make the documentation of capitalization complete and yet make it stand on its own, that would lead to a lot of repetition. Capitalization is really just a special case of emphasis and I think the documentation should express that. I do think the user should be able to roughly understand what the "caps" opcodes do without having to jump to other sections. For that reason I suggested to include a brief explanation of the opcode. But for the fine details one would have to read the "emph" counterpart.

bertfrees commented:
I have cleaned up the branch a bit. I am planning the merge the first 4 commits already and then we can think some more about the caps stuff.

@egli You approved the PR, but you had several comments. Can you check that they have been addressed?

BueVest commented:
I think I addressed all the comments from @egli, but if I have forgotten anything, please tell me.

 

Concerning caps etc.,I tested the various caps features some more, but can’t really get them to work as I expected. If we require tests from table authors, there might be something to say for requiring documentation when new features are implemented.

bertfrees commented:
What is it that does not work as expected?

We are already providing documentation for all new features.

egli commented:
I pushed a few slight texinfo cleanups (put standing alone consistently in emph and drop the emph from the node name, as node names should not contain any macros as far as I know)

So as far as I'm concerned we can merge.

egli commented:
The only thing remaining seems to be the [discussion about the seqdelimiter in the yaml test](https://github.com/liblouis/liblouis/pull/928#pullrequestreview-396355294) .

bertfrees commented:
@egli Thanks. Merged everything except the begcaps/endcaps commit and the capitalization.yaml commit: e4a3793.

egli commented:
@bertfrees thanks for merging. I guess we can close this PR then
